### ISSUE TITLE
Manual install on a long running raspiblitz: Error processing background data: list index (0) out of range

### ISSUE THREAD

**Issue by openoms** [cryptosharks131/lndg#50]
Started to test LNDg as hearing more about it and got interested. Nice project!

Find my notes about the installation to a raspiblitz with a dedicated user:
https://github.com/openoms/lightning-node-management/blob/en/hardware/raspiblitz/lndg.md

However the installation and the interface are running fine I get and error with the jobs:
```
$ sudo -u lndg /home/lndg/lndg/.venv/bin/python /home/lndg/lndg/jobs.py
Error processing background data: list index (0) out of range
```

Background info about the node:
```
₿ uname -a
Linux raspberrypi 5.10.63-v8+ #1496 SMP PREEMPT aarch64 GNU/Linux
₿ python --version
Python 3.7.3
₿ /home/lndg/lndg/.venv/bin/pip --version
pip 22.0.3 from /home/lndg/lndg/.venv/lib/python3.7/site-packages/pip (python 3.7)
₿ lnd --version
lnd version 0.14.2-beta commit=v0.14.2-beta
````

it has some known quirks in the database including stuck channels and some issue with recalling the payment history. 
Running LND v0.14.2, but has been through many versions and channels (120+ closed) over 2+ years.

Happy to test if there are some suggestions to make LNDg work,but wouldn't be surprised if it is a unique issue related to the node's database. 
Documenting also if any others would have a similar issue. Will test on a cleaner raspiblitz and on FreeBSD too (for that great that supervisord can be used).

**Comment by cryptosharks131** [comment]
Wow, an honor to have you here @openoms and thank you!

A few issues were addressed in the data processing in next version v1.0.4 including:
-When channel ids were outside the signed integer range
-When you have opened private channels

Both these issues have been resolved in v1.0.4 and it should be ready for release soon. If you would like, you can clear your data folder inside LNDg and then switch to the v1.0.4 branch to see if you are then able to load data successfully.

**Comment by openoms** [comment]
Thanks for the reply. 
Switched to the latest commit in `v1.0.4`. Now it seems to manage to do more work, but eventually fails with the same:
```
/home/lndg/lndg(v1.0.4) ₿ time sudo -u lndg /home/lndg/lndg/.venv/bin/python /home/lndg/lndg/jobs.py
Error processing background data: list index (0) out of range

real	0m16.074s
user	0m2.455s
sys	0m0.407s
```

**Comment by cryptosharks131** [comment]
Thanks for trying that, the only issue I can see where that might happen now is in the invoice records and potentially missing some data in the gRPC response.

If you could remove the try/except code from this code block, the error should give a more robust trace back, hopefully, pointing to the line where the issue is occuring.

https://github.com/cryptosharks131/lndg/blob/v1.0.4/jobs.py#L339-L352

**Comment by openoms** [comment]
Now getting:
```
Traceback (most recent call last):
  File "/home/lndg/lndg/jobs.py", line 355, in <module>
    main()
  File "/home/lndg/lndg/jobs.py", line 345, in main
    update_invoices(stub)
  File "/home/lndg/lndg/jobs.py", line 99, in update_invoices
    alias = Channels.objects.filter(chan_id=invoice.htlcs[0].chan_id)[0].alias if Channels.objects.filter(chan_id=invoice.htlcs[0].chan_id).exists() else None
IndexError: list index (0) out of range
```

**Comment by cryptosharks131** [comment]
Ah that is the spot I thought might be the issue. It looks like you may have an older invoice without the incoming HTLC data attached which tells us on which channel the invoice was received.

**Comment by cryptosharks131** [comment]
Original issued described by OP resolved in #51
